### PR TITLE
TF-11619/apply link scan stack to cdk app base template

### PR DESCRIPTION
**Pull request by e-papanicolas**
### Ticket(s)

-   _[11619](https://airtable.com/appfJZShN8K4tcWHU/pagHwvtNNoIIuXqAz?GOswY=recG3YBGwnBxrQO5z)_

### Type of change

-   [x] New application

### Description

See `/lambda/link-scan/README.md` for the project details. This PR also includes a lot of setup. The initial commit of this repo was just the base code when running `cdk app typescript` which Ccreates an empty AWS CDK app. 

When we have the CDK section of our AWS documentation complete I can add an overview of it to the main README for our CDK app.

Please leave feedback or suggestions for any improvement, especially things like file structure and file names which could use a second opinion.

Beyond the last review I updated the use of the typescript SDK from v2 to v3, used in the implementation of SES in `/lambda/link-scan/index.ts`, I added tests with `vitest`, added configuration files for things like CODEOWNERS, eslint, and dependabot.

#### Code Clarity and Readability

-   [x] Are variable and function names descriptive, meaningful, in the appropriate case (ex: snake, camel)?
-   [x] Is the code self-explanatory and easy to understand?
-   [x] Are comments used where necessary to explain complex logic or provide additional context?

#### Code Efficiency

-   [x] Is the code DRY?
-   [x] Are there any areas of unnecessary complexity or duplication that might benefit from refactoring?

#### Code Modularity and Reusability

-   [x] Is the code and any new components organized in a logical location in the directory?
-   [x] Can any code snippets be refactored into reusable components?
-   [x] Are there opportunities to abstract common functionality into separate functions or classes?

#### Best Practices and Coding Standards

-   [x] Does the code follow the established coding standards and style guide of the project?
-   [x] Are there any violations of best practices that should be addressed?
-   [x] Is there any unnecessary commented out code lingering?

### Overall PR Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have created a readme if this is a new project
-   [x] My changes generate no new warnings in the console
-   [x] There are no new linting errors/problems in my code
-   [x] I have added tests that prove my application is effective or that my feature works
-   [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

File: lib/link-scan/link-scan-stack.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        super(scope, id, props)
+
+        const linkScanFn = new NodejsFunction(this, 'LambdaHandler', {
+            entry: path.join(__dirname, '../../lambda/link-scan/index.ts'),
+            handler: 'handler',
+            runtime: Runtime.NODEJS_18_X,
+            timeout: Duration.seconds(300),
+        })
+
+        linkScanFn.addToRolePolicy(
```
Comment (jteitel10) [inline lib/link-scan/link-scan-stack.ts]: Alternatively, you could just create an [initialPolicy](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.NodejsFunction.html#initialpolicy)

**Review by jteitel10 — COMMENTED** [review]
Looks good. Instead of add to role policy for the Lambda, you could just use an [initial policy](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.NodejsFunction.html#initialpolicy) parameter

File: lib/link-scan/link-scan-stack.ts (hunk 1 above)
Comment (e-papanicolas) [reply lib/link-scan/link-scan-stack.ts]: That is much nicer to look at! Thanks 👍

File: lambda/link-scan/index.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            brokenLinks.push(newBrokenLink)
+        }
+    })
+
+    const result = await checker.check({
+        path: URL_TO_SCAN,
+        recurse: true,
+    })
+
+    const brokenLinksCount = result.links.filter((x) => x.state === 'BROKEN')
```
Comment (LeoFuna) [inline lambda/link-scan/index.ts]: Just a little thing, what about change this `x` var name?
We could try make this more specific about on what we are talking about

**Review by sramalho94 — APPROVED** [review]
@e-papanicolas amazing work, I especially love the tests you made

**Pull request merged**